Blackbaud.CustomFx.ConstituentPhotoAnalysis.Catalog/ConstituentPhotoAnalysis.View.xml

```diff
@@ -0,0 +1,74 @@
+﻿<ViewDataFormTemplateSpec
+  xmlns="bb_appfx_viewdataformtemplate"
+  xmlns:c="bb_appfx_commontypes"
+  ID="13e631e3-a570-4f95-a65d-653a1020418e"
+  Name="Constituent Photo Analysis View Form"
+  Description="A data form for viewing the results of a constituent photo analysis."
+  Author="Blackbaud Demo"
+  DataFormInstanceID="c8eb35c2-6fda-4f1e-a379-b7609aca772f"
+  RecordType="Constituent"
+  c:SecurityUIFolder="Constituent Photo Analysis"
+  >
+
+  <SPDataForm SPName="USR_USP_DATAFORMTEMPLATE_VIEW_CONSTITUENTPHOTOANALYSIS">
+    <c:CreateProcedureSQL>
+      <![CDATA[
+create procedure dbo.USR_USP_DATAFORMTEMPLATE_VIEW_CONSTITUENTPHOTOANALYSIS
+(
+  @ID uniqueidentifier,
+  @DATALOADED bit = 0 output,
+  @PREDICTEDGENDER nvarchar(7) = null output,
+  @ESTIMATEDAGE int = null output,
+  @ANGERRATING  decimal(20, 4) = null output,
+  @CONTEMPTRATING  decimal(20, 4) = null output,
+  @DISGUSTRATING  decimal(20, 4) = null output,
+  @FEARRATING  decimal(20, 4) = null output,
+  @HAPPINESSRATING  decimal(20, 4) = null output,
+  @NEUTRALRATING  decimal(20, 4) = null output,
+  @SADNESSRATING  decimal(20, 4) = null output,
+  @SURPRISERATING  decimal(20, 4) = null output
+)
+as
+  set nocount on;
+
+  set @DATALOADED = 0;
+	
+  select 
+    @DATALOADED = 1,
+    @PREDICTEDGENDER = PREDICTEDGENDER,
+    @ESTIMATEDAGE = ESTIMATEDAGE,
+    @ANGERRATING = ANGERRATING,
+    @CONTEMPTRATING = CONTEMPTRATING,
+    @DISGUSTRATING = DISGUSTRATING,
+    @FEARRATING = FEARRATING,
+    @HAPPINESSRATING = HAPPINESSRATING,
+    @NEUTRALRATING = NEUTRALRATING,
+    @SADNESSRATING = SADNESSRATING,
+    @SURPRISERATING = SURPRISERATING
+  from 
+    dbo.CONSTITUENT
+    left join dbo.USR_CONSTITUENTPHOTOANALYSIS on USR_CONSTITUENTPHOTOANALYSIS.ID = CONSTITUENT.ID
```

Comment:
Can we eliminate the join by selecting directly from the USR_CONSTITUENTPHOTOANALYSIS table where USR_CONSTITUENTPHOTOANALYSIS.ID = @ID? I see the select statement contains no CONSTITUENT table column data.

Reply:
The constituent join exists to prevent a "record not found" error on the tab. I could've also conditionally shown the tab based on the existence of the record, but that would've required modifying the constituent page view form.